code/modules/wiremod/components/action/speech.dm

```diff
@@ -31,7 +31,7 @@
 		var/atom/movable/shell = parent.shell
 		// Prevents appear as the individual component if there is a shell.
 		if(shell)
-			shell.say(message.value)
+			shell.say(message.value, forced = "circuit speech | Built by [key_name(parent.inserter_mind.resolve())]")
 		else
-			say(message.value)
```

Comment:
you can use get_creator instead to list who inserted the circuit, which id the circuit is locked to and the shell name